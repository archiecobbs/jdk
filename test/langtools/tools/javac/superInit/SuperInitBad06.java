/*
 * @test /nodynamiccopyright/
 * @bug 8194743
 * @summary Test invalid reference to "this" prior to superclass construction
 *
 * @compile/fail/ref=SuperInitBad06.out -XDrawDiagnostics SuperInitBad06.java
 */
class SuperInitBad06 {

    int x;

    SuperInitBad06() {
        this.x++;
        super();
    }
}
