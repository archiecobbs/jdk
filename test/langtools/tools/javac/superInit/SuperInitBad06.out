SuperInitBad06.java:13:9: compiler.err.cant.ref.before.ctor.called: this
1 error
